src/Compilers/CSharp/Portable/Binder/Binder_Invocation.cs

```diff
@@ -1141,11 +1141,28 @@ private void CheckRestrictedTypeReceiver(BoundExpression expression, CSharpCompi
 
             return new BoundCall(node, receiver, method, args, argNames, argRefKinds, isDelegateCall: isDelegateCall,
                         expanded: expanded, invokedAsExtensionMethod: invokedAsExtensionMethod,
+                        receiverCloned: ReceiverCloned(receiver, method),
                         argsToParamsOpt: argsToParams, defaultArguments, resultKind: LookupResultKind.Viable, type: returnType, hasErrors: gotError);
         }
 
 #nullable enable
 
+        internal bool ReceiverCloned(BoundExpression? receiver, PropertySymbol property)
+            => ReceiverCloned(receiver, property.GetMethod ?? property.SetMethod);
+
+        internal bool ReceiverCloned(BoundExpression? receiver, MethodSymbol method)
+        {
+            if (receiver is BoundValuePlaceholderBase || receiver?.Type?.IsValueType != true)
+            {
+                return false;
+            }
+
+            var valueKind = method.IsEffectivelyReadOnly
+                ? BindValueKind.RefersToLocation
+                : BindValueKind.RefersToLocation | BindValueKind.Assignable;
+            return !CheckValueKind(receiver.Syntax, receiver, valueKind, checkingReceiver: true, BindingDiagnosticBag.Discarded);
```

Comment:
>BindingDiagnosticBag.Discarded [](http://example.com/codeflow?start=97&length=30)

Since we moved this check to the initial binding, it looks like it will be more robust to pass real BindingDiagnosticBag from the caller.